### ISSUE TITLE
Defining a DataAdapter with amd

### ISSUE THREAD

**Issue by RobertDiebels** [select2/select2#4153]
Hi,

I've got a question concerning DataAdapters I don't know if this is an issue with my code or with select2 so I'm posting it here instead of stackoverflow.

We don't use anything like amd or requirejs atm so I'm currently using almond (which select2 provides).

```js
var self = this;
field.select2({
    data:        _.map(self.data, function ( item ) {
        return {
            id:   item.id,
            text: item.name
        };
    }),
    dataAdapter: $.fn.select2.amd.require('CustomDataAdapter')
```

And I've defined a DataAdapter as follows:

```js
$.fn.select2.amd.define('CustomDataAdapter',[
    'select2/data/array',
    'select2/utils'
], function (ArrayData, Utils) {
    function CustomDataAdapter ($element, options) {
        CustomDataAdapter.__super__.constructor.call(this, $element, options);
        console.log("HERPDERP",$element, options);
    }

    Utils.Extend(CustomDataAdapter, ArrayData);

    CustomDataAdapter.prototype.query = function (params, callback) {
        console.log("HERPDERP2");


        var data = {
            results: []
        };

        for (var i = 1; i < 5; i++) {
            var s = "";

            for (var j = 0; j < i; j++) {
                s = s + params.term;
            }

            data.results.push({
                id: params.term + i,
                text: s
            });
        }

        callback(data);
    };

   return CustomDataAdapter;
});
```

I can see the call to the constructor in the console. But query() never gets called. Does this have something to do with my approach? Or am I missing something else?

**Comment by kevin-brown** [comment]
As mentioned by @AubreyHewes, `query` is only called when the results need to be loaded.

---

This appears to be more of a usage question than a bug report or common feature request.  As we continue to keep the Select2 issue tracker geared towards these two areas, bug reports and common feature requests, we are redirecting usage questions to other communities such as the [mailing list][ml], [IRC][irc], or [Stack Overflow][so].

[ml]: https://github.com/select2/select2#mailing-list
[irc]: https://github.com/select2/select2#irc-channel
[so]: http://stackoverflow.com/questions/tagged/jquery-select2?sort=votes

**Comment by RobertDiebels** [comment]
@AubreyHewes  Obviously I'm using v4 because I'm getting a console.log output as I've mentioned above. Also I've clicked the box a multitude of times before I submitted a bug-report and nothing happened.
I'm using select2 in a bootstrap modal. Which might cause the problems. 

This is not a usage problem as far as I can see. The code I'm using is as provide above.
With the addition that it's used inside a bootstrap modal.
